#441 part 3 reviewer comments on summary

nmadruga opened the pull request:
Fixes #441

### Changes
- **Warning**: This PR does a lot more than what should be included. The reason is trying to get most of of was included on the demo for style layout of reviews on the Application Summary and Review Page.
- Styles are **inline** to be moved to LESS files and using classes when we start changing global styles on Epic#10
- Modified `SummaryViewWrapperNew` to display content **inline**, not sure if that's desired, but giving a try.
This is how its looking on a new application:
<img width="933" alt="Screen Shot 2021-04-08 at 11 03 51 AM" src="https://user-images.githubusercontent.com/16461988/113945285-2890f800-985a-11eb-834f-9c9c351c6ac7.png">


- Modified `ReviewPage`, `ApplicationSummary`, `SectionWrapper` and `PageElements` to show the Applicant current response, the review and (in ApplicationSummary) besides the current response, also the previous response and reviewer comments.
**[EDITED - 10/04/21]**
- Added components for each block of Response and Review displayed as part of `PageElements` and store in new folder `components > Application > PageElements > Elements`. 
- Used variables to keep inlineStyles separated from each component

This is how shows on the `ReviewPage` (**[UPDATED - 10/04/21]**):
<img width="723" alt="Screen Shot 2021-04-10 at 10 20 57 PM" src="https://user-images.githubusercontent.com/16461988/114266624-59844f00-9a4b-11eb-9a9a-f2140abdfc27.png">

This is how is showing on the `ApplicationSummary` after the applicant has made the **changes required** (**[UPDATED - 10/04/21]**):
1
<img width="850" alt="Screen Shot 2021-04-10 at 10 27 54 PM" src="https://user-images.githubusercontent.com/16461988/114266757-1080ca80-9a4c-11eb-8ca6-fb3f6aa932ea.png">

2
<img width="735" alt="Screen Shot 2021-04-10 at 10 28 06 PM" src="https://user-images.githubusercontent.com/16461988/114266762-14ace800-9a4c-11eb-9aaa-d7dcd000682c.png">

### Unrelated changes:
- Changed `SiteLayout`, `Footer`, `UserArea`, `AuthenticatedWrapper` to use fixed positions for anything besides what is currently displayed on the main area (of SiteLayout). This allowed the sticky features to work correctly (almost - see problem related to that [here](#254)).
- Display dates in **string** format on `LastActiveDateCell` and `PageElements`
- Moved `AppMenu` and `UserSelection` to separated component - displayed on the right hand for development `DevOptions`. There is one tiny issue with it described in the comments.
New menu on the right-hand side:
<img width="149" alt="Screen Shot 2021-04-08 at 10 40 06 AM" src="https://user-images.githubusercontent.com/16461988/113945326-3ba3c800-985a-11eb-97b7-ff5e2e24a39a.png">

andreievg commented on semantic/src/site/globals/site.overrides:
```diff
@@ -1,3 +1,38 @@
+                color: #003BFE
+}
+
+.sup .ui.tiny.progress {
```
oops, my play around class names are not very professional `sup` and `dreistripes` lol, would be good to change to something more appropriate

andreievg replied:
although stripes are probably not used in this pr

nmadruga replied:
Yes, I just moved the whole file to here - I'll do another PR for changes to the list (hopefully) today and will branch from this one so will keep this here but renamed.

andreievg commented on src/components/Main/AppMenu.tsx:
```diff
@@ -1,41 +1,76 @@
-        ))}
-      </Menu>
-    )
+    <Popup
```
I think we should move all of the dev components to 'dev' folder (under containers ? )

nmadruga replied:
No problem. Done: c8dee56

andreievg commented on src/containers/Application/ApplicationSummary.tsx:
```diff
@@ -69,24 +69,81 @@ const ApplicationSummary: React.FC<ApplicationProps> = ({
+          as="h1"
+          textAlign="center"
+          style={{ fontSize: 26, fontWeight: 900, letterSpacing: 1, marginBottom: 4 }}
+          content="Review and submit"
```
localisation

andreievg commented on src/containers/Application/ApplicationSummary.tsx:
```diff
@@ -69,24 +69,81 @@ const ApplicationSummary: React.FC<ApplicationProps> = ({
+          style={{ marginTop: 4, color: '#4A4A4A', fontSize: 16, letterSpacing: 0.36 }}
+          as="h3"
+        >
+          Please review each section before submitting form
```
localisation

andreievg reviewed (changes requested):
I am trying to figure out how to keep the 'polution' from demo branch away from master.  I  think, if we are going to be bringing in styles from that branch (vs doing styles in strategic way), we should think about how to manage that polution.

How about extracting all of the inline styles into variables.

for example `SectionWrapper.tsx`, at the top or bottom:

```
const inlineStyles = {
  mainAccordion: {
      borderRadius: 8,
      marginBottom: 10,
      border: 'none',
      boxShadow: 'none',
      backgroundColor: '#DCDDDD'
  },
 titleGrid: {
      margin: 0,
      borderRadius: 8,
      backgroundColor: '#DCDDDD',
 },
..etc
}
```

and then use them in jsx.

I know it's a bit more work, but I am a bit concerned about inline styles like this in master.

Otherwise PR looks good

nmadruga commented:
> How about extracting all of the inline styles into variables.

Sure thing - I'll start doing that now

nmadruga commented:
Finally ready for re-review. Sorry, it took me longer to figure out some stuff and end up moving components (to render single Response, Review, Previous response, etc to separated files - on new folder `PageElements`.
I believe I have addressed the remaining suggestions, but now there are more files changes. So, please have a look.

nmadruga commented:
The problem with the new `DevOptions` on the right is to be collapsing with the Main component on the right side (see images in the description where the right side has a strange border) the problem gets worse on mobile view.

andreievg commented on src/components/Application/Sections/SectionWrapper.tsx:
```diff
@@ -0,0 +1,113 @@
+  sup: {
+    borderRadius: 8,
+    marginBottom: 10,
+    border: 'none',
```
padding: 10, would help here:

![Screen Shot 2021-04-12 at 2 11 52 PM](https://user-images.githubusercontent.com/26194949/114331941-4089b400-9b99-11eb-87b4-1078840c8b9a.png)

vs

![Screen Shot 2021-04-12 at 2 14 17 PM](https://user-images.githubusercontent.com/26194949/114332013-6747ea80-9b99-11eb-992e-3e49c62232da.png)

nmadruga replied:
Better. Thanks 6468715

andreievg reviewed (changes requested):
a few extra changes.

I am not really sure about the relocation of page elements. This would require quite a bit more reviewing and testing (I think we need to be careful, when a couple of line PR for adding comments goes this far, was happy to move some styles but not sure about reshuffling of page elements, especially since there is also consolidation to be done yet into page elements, which would have been a good time to do the reshuffle once we can see all common components). Anyways I'll review this more thoroughly, but it would take a bit more time.

nmadruga commented:
> I am not really sure about the relocation of page elements. This would require quite a bit more reviewing and testing (I think we need to be careful, when a couple of line PR for adding comments goes this far, was happy to move some styles but not sure about reshuffling of page elements, especially since there is also consolidation to be done yet into page elements, which would have been a good time to do the reshuffle once we can see all common components). Anyways I'll review this more thoroughly, but it would take a bit more time.

I see your point. Sorry you feel more lost with where things are, I tried not to change how PageElements is structured, just moved components that were inside it to separated files. Interesting that for me with all the components in the same file was starting to get way convoluted for me to set styles, I kept losing where I was. That's  why I moved to separated files...

nmadruga commented on src/components/Main/Footer.tsx:
```diff
@@ -1,63 +1,27 @@
-      </List>
-    </Container>
-  </Segment>
+  <Container style={inlineStyles}>
```
Yes, missed those ones. I was trying to use some smart mapping on the children of `inlineStyles` before, but that wasn't working well, so moved to call each field as you suggested here too

nmadruga replied:
Done: 41389aa

nmadruga commented on src/containers/Dev/DevOptions.tsx:
```diff
@@ -0,0 +1,26 @@
+
+  const { filteredTemplates } = useListTemplates(templatePermissions, isLoading)
+  return (
+    <Menu vertical size="tiny" fixed="right" floated="right" style={{ zIndex: 0 }}>
```
Very nice! Problem described [here](https://github.com/openmsupply/application-manager-web-app/pull/565#issuecomment-817114907) fixed! Done: c7f3047

CarlosNZ commented on src/components/Main/Footer.tsx:
```diff
@@ -1,63 +1,27 @@
+// Styles - TODO: Move to LESS || Global class style (semantic)
+const inlineStyles = {
+  container: {
+    position: 'fixed',
```
This seems weird to be "fixed". It's just taking up screen real estate for no reason.

nmadruga replied:
This was done to help with the `Sticky` elements (running inside the main area - `SiteLayout`). Testing now it doesn't  seem to affect the Sticky elements, so I'm removing this `fixed` property here: 08ff9ea

CarlosNZ commented on src/containers/Main/SiteLayout.tsx:
```diff
@@ -19,88 +19,108 @@ import {
 
+// Styles - TODO: Move to LESS || Global class style (semantic)
+const containerStyle = {
+  minHeight: '100vh',
```
Not sure why we need this. Once you turn the Footer off "fixed" (my other suggestion), it disappears because of this. Maybe we don't need it at all, but if we want it, then I wouldn't go more than 90vh.

nmadruga replied:
Done: e86b53a

CarlosNZ reviewed (commented):
Sorry for jumping in here at the last minute. This is looking really good, I just made a couple of comments on things I found odd.

Also, this might be beyond what you want to do here, but the Navigation bar (Next/Prev) within applications is looking a bit weird -- making it match the width of the container would help.

nmadruga commented:
> Also, this might be beyond what you want to do here, but the Navigation bar (Next/Prev) within applications is looking a bit weird -- making it match the width of the container would help

Oh, you are completely right. I forgot about moving styles from `demo-22-03` for that one. I'll do this now. Thanks!

Will also update this branch with master to fix conflicts

nmadruga commented:
@andreievg & @CarlosNZ please have another look onto this PR, hopefully we can merge this one sooner than later so we apply styles on top of it and should be easier to deal with my next PR (all based from this one).

CarlosNZ commented on src/components/Application/PageElements/Elements/SummaryReviewResponseElement.tsx:
```diff
@@ -22,14 +22,15 @@ const SummaryReviewResponseElement: React.FC<SummaryReviewResponseElementProps>
   if (reviewResponse?.decision === ReviewResponseDecision.Approve) return null
 
-  const { element, allResponses } = summaryProps
+  const { element, allResponses, applicationData } = summaryProps
```
Great, thanks for updating these in here.

CarlosNZ reviewed (approved):
All looks good now, thanks :)

andreievg reviewed (approved):
Looks good Nicole, thanks for the changes
